pyfar/dsp/dsp.py

```diff
@@ -1386,6 +1386,241 @@ def time_shift(signal, shift, unit='samples'):
+        >>> ir = pf.signals.impulse(n_samples)
+        >>> ir = pf.dsp.linear_phase(ir, delay_samples, unit='samples')
+        >>> start_samples = pf.dsp.find_impulse_response_delay(ir)
+        >>> ax = pf.plot.time(ir, unit='ms', label='impulse response')
+        >>> ax.axvline(
+        ...     start_samples/ir.sampling_rate*1e3,
+        ...     color='k', linestyle='-.', label='start sample')
+        >>> ax.legend()
+
+    """
+    n = int(np.ceil((N+2)/2))
+
+    start_samples = np.zeros(impulse_response.cshape)
+    for ch in np.ndindex(impulse_response.cshape):
+        # Calculate the correlation between the impulse response and its
+        # minimum phase equivalent. This requires a minimum phase equivalent
+        # in the strict sense, instead of the appriximation implemented in
+        # pyfar.
+        n_samples = impulse_response.n_samples
+        ir_minphase = sgn.minimum_phase(
+            impulse_response.time[ch], n_fft=4*n_samples)
+        correlation = sgn.correlate(
+            impulse_response.time[ch],
+            np.pad(ir_minphase, (0, n_samples - (n_samples + 1)//2)),
+            mode='full')
+        lags = np.arange(-n_samples + 1, n_samples)
+
+        # calculate the analytic signal of the correlation function
+        correlation_analytic = sgn.hilbert(correlation)
+
+        # find the maximum of the analytic part of the correlation function
+        # and define the search range around the maximum
+        argmax = np.argmax(np.abs(correlation_analytic))
+        search_region_range = np.arange(argmax-n, argmax+n)
+        search_region = np.imag(correlation_analytic[search_region_range])
+
+        # mask values with a negative gradient
+        mask = np.gradient(search_region, search_region_range) > 0
+
+        # fit a polygon and estimate its roots
+        search_region_poly = np.polyfit(
+            search_region_range[mask]-argmax, search_region[mask], N)
+        roots = np.roots(search_region_poly)
+
+        # Use only real-valued roots
+        if np.all(np.isreal(roots)):
+            root = roots[np.abs(roots) == np.min(np.abs(roots))]
+            start_sample = lags[argmax] + root
+        else:
+            start_sample = np.nan
+            warnings.warn(f"Starting sample not found for channel {ch}")
+
+        start_samples[ch] = start_sample
+
+    return start_samples
+
+
+def find_impulse_response_start(
+        impulse_response,
+        threshold=20):
+    """Find the start sample of an impulse response.
+
+    The start sample is identified as the first sample which is below the
+    ``threshold`` level relative to the maximum level of the impulse response.
+    For room impulse responses, ISO 3382 [#]_ specifies a threshold of 20 dB.
+
+    .. note::
+
+        This function is primary intended to be used when processing room
+        impulse responses.
+        Alternatively see :py:func:`pyfar.dsp.find_impulse_response_delay`.
```

Comment:
I'd say a note is not necessary here and adding this just as a sentence is fine

Reply:
:+1: